symbolic-symcache/src/new/writer.rs

```diff
@@ -182,11 +158,8 @@ impl SymCacheConverter {
     pub fn process_symbolic_function(&mut self, function: &Function<'_>) {
         let comp_dir = std::str::from_utf8(function.compilation_dir).ok();
 
-        let entry_pc = if function.inline {
-            u32::MAX
-        } else {
-            function.address as u32
-        };
+        // Is this correct for inlined functions?
+        let entry_pc = function.address as u32;
```

Comment:
maybe add a comment that we are doing this purely for the reason to keep the changes to the snapshot of `functions` minimal. For inlined functions this indeed does not make sense.

But consider this:

We have a function which exists *both* as a freestanding toplevel function, *and* as an inlined function.
IMO, the `function_idx` would point to the toplevel function in that case, similar to how `DW_AT_abstract_origin` works.

Additional metadata which might be interesting for the future would be: file/line where this function is defined. We don’t expose that at the moment, but might in the future. That info would be the same no matter if the function is being inlined or not.

On the other hand, we save generic instanciations separately. So `a<u32>` would create a different function record from `a<u64>`, they have different `entry_pc` (if they are freestanding), but the same declaration file/line.